packages/core/contracts/oracle/implementation/Staker.sol

```diff
@@ -35,6 +35,7 @@ contract Staker is StakerInterface, Ownable {
         uint256 outstandingRewards;
         uint64 lastRequestIndexConsidered;
         uint64 unstakeRequestTime;
+        uint64 unstakeCoolDown;
```

Comment:
the main downside with this is we definitionally increase the gas cost of all operations by doing this, which kind of sucks. this also increases the bytecode of the contract for arguably somewhat minimal gain.

I do think, though, that this is worth it given we get best of both worlds: users know that they will, at worst, get the unstake timer they requested when unstaking and this can be short circuited by a vote to decrease the `unstakeCooldown` for all stakers, in the event of a migration or whatever.

Reply:
Have we considered leaving this as is?

It does mean you have to trust the DAO not to try to lock users' funds. However, the stakers would have to vote on this! This is not a multisig owner.

There is definitely a case where a user has their tokens forcibly locked during their withdrawal period, but the DAO doesn't have much incentive to do something like this as it implicitly locks the other staker's funds as well.

I could see this potentially being more important for usability/UX as a user might be annoyed if they are unstaking and the wait period increases _after_ they have locked into removing their tokens. However, I think this really depends on how _often_ we think the system will be changing this parameter.

Not super strongly opinionated either way, just wanted to mention the option of keeping this as is since it does add some cost and complexity.